Formula/sip.rb

```diff
@@ -32,18 +33,30 @@ class Sip < Formula
     sha256 "b3bda1d108d5dd99f4a20d24d9c348e91c4db7ab1b749200bded2f839ccbe68f"
   end
 
+  # TODO: remove them after sip 6.1.0
+  # These patch provide the option `--scripts-dir`
+  patch do
+    url "https://www.riverbankcomputing.com/hg/sip/raw-diff/ffd0551c32cc/sipbuild/builder.py"
+    sha256 "2c969dfba2e4b0553d06999a3aa07a93ea4b7ca2cce62635d1418ecdc74a6df2"
+  end
+
+  patch do
+    url "https://www.riverbankcomputing.com/hg/sip/raw-diff/ffd0551c32cc/sipbuild/project.py"
+    sha256 "ea99834ab404583a1a49a05997950758fe95ba473129438117ffa5647028d99a"
+  end
+
   def install
-    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
-    venv = virtualenv_create(libexec, Formula["python@3.9"].opt_bin/"python3")
-    %w[packaging pyparsing toml].each do |r|
+    python = Formula["python@3.9"]
+    venv = virtualenv_create(libexec, python.bin/"python3")
+    resources.each do |r|
       venv.pip_install resource(r)
     end
 
-    system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(prefix)
+    system python.bin/"python3", *Language::Python.setup_install_args(prefix)
 
-    site_packages = libexec/"lib/python#{xy}/site-packages"
-    pth_contents = "import site; site.addsitedir('#{site_packages}')\n"
-    (lib/"python#{xy}/site-packages/homebrew-sip.pth").write pth_contents
+    site_packages = Language::Python.site_packages(python)
+    pth_contents = "import site; site.addsitedir('#{libexec/site_packages}')\n"
+    (prefix/site_packages/"homebrew-sip.pth").write pth_contents
```

Comment:
I know this is an issue with the previous version of the formula too. But doesn't mean that when I install `sip`, then `sip`'s resources will be exposed to the global Python environment?

That is, when I do `brew install sip` and then `pip3 list`, I'm going to see `packaging`, `pyparsing` and `toml` listed there even if I never installed them there (because `sip` installed them there for me).

Do we want that? Won't it potentially conflict with other Python formulae?

Reply:
Or, wait, maybe your changes here actually fixes it.